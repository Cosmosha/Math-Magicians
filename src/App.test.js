import renderer from 'react-test-renderer';
import operate from './logic/operate';
import calculate from './logic/calculate';
import Calculator from './components/calculator';
import Home from './components/Home';
import Quote from './components/quote';

describe('Test calculate', () => {
  it('6 + 4 = 10', () => {
    const obj = {
      total: '6',
      next: '4',
      operation: '+',
    };
    const res = calculate(obj, '=');
    expect(res.total).toBe('10');
  });

  it('6 - 4 = 2', () => {
    const obj = {
      total: '6',
      next: '4',
      operation: '-',
    };
    const res = calculate(obj, '=');
    expect(res.total).toBe('2');
  });

  it('6 x 4 = 24', () => {
    const obj = {
      total: '6',
      next: '4',
      operation: 'x',
    };
    const res = calculate(obj, '=');
    expect(res.total).toBe('24');
  });

  it('6 รท 4 = 1.5', () => {
    const obj = {
      total: '6',
      next: '4',
      operation: 'รท',
    };
    const res = calculate(obj, '=');
    expect(res.total).toBe('1.5');
  });

  it('6 % 4 = 2', () => {
    const obj = {
      total: '6',
      next: '4',
      operation: '%',
    };
    const res = calculate(obj, '=');
    expect(res.total).toBe('2');
  });
});

describe('Testing Operate', () => {
  it('6 + 4 = 10', () => {
    const result = operate(6, 4, '+');
    expect(result).toBe('10');
  });
  it('6 - 4 = 2', () => {
    const result = operate(6, 4, '-');
    expect(result).toBe('2');
  });
  it('6 x 4 = 24', () => {
    const result = operate(6, 4, 'x');
    expect(result).toBe('24');
  });
  it('6 รท 4 = 1.5', () => {
    const result = operate(6, 4, 'รท');
    expect(result).toBe('1.5');
  });
  it('6 % 4 = 2', () => {
    const result = operate(6, 4, '%');
    expect(result).toBe('2');
  });
});

describe('Component UI render as expected', () => {
  it('Calculator renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Home renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Quote renders correctly', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
