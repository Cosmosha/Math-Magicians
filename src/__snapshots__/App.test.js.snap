// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test calculate Component UI render as expected Calculator renders correctly 1`] = `
Array [
  <div
    className="container"
  >
    <h1
      className="headline"
    >
       Let do some math! 
    </h1>
     
    <div
      className="calculator-grid"
    >
      <div
        className="display"
      >
        <div
          className="previou-operand"
        >
          0
           
        </div>
        <div
          className="current-operand"
        />
      </div>
      <button
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
         +/- 
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
         % 
      </button>
      <button
        className="operators"
        onClick={[Function]}
        type="button"
      >
         รท 
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
         7 
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
         8 
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
         9 
      </button>
      <button
        className="operators"
        onClick={[Function]}
        type="button"
      >
         x 
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
         4 
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
         5 
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
         6 
      </button>
      <button
        className="operators"
        onClick={[Function]}
        type="button"
      >
         - 
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
         1 
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
         2 
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
         3 
      </button>
      <button
        className="operators"
        onClick={[Function]}
        type="button"
      >
         + 
      </button>
      <button
        className="span-two"
        onClick={[Function]}
        type="button"
      >
         0 
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
         . 
      </button>
      <button
        className="operators"
        onClick={[Function]}
        type="button"
      >
         = 
      </button>
    </div>
  </div>,
  " ",
]
`;

exports[`Test calculate Component UI render as expected Home renders correctly 1`] = `
<div
  className="quote-para"
>
  <h1>
     Welcome to Our Page! 
  </h1>
   
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
     
  </p>
   
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmodd tempor incididunt ut labore et dolore magna aliqua.Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
     
  </p>
   
</div>
`;

exports[`Test calculate Component UI render as expected Quote renders correctly 1`] = `
<ul
  className="getFetched"
>
   
   
</ul>
`;
